.gitlab-ci.yml

```diff
@@ -6,30 +6,169 @@ image:
   name: mcr.microsoft.com/dotnet/sdk:6.0
   entrypoint: [""]
 
-cache:
-  key: "$CI_JOB_STAGE-$CI_COMMIT_REF_SLUG"
-  paths:
-    - "./packages/pangea-sdk/PangeaCyber.Net/obj/project.assets.json"
-    - "./packages/pangea-sdk/PangeaCyber.Net/obj/*.csproj.nuget.*"
-    - ".nuget"
-  policy: pull-push
-
-before_script:
-  - cd packages/pangea-sdk
-  - dotnet restore --packages .nuget
-  - dotnet build --no-restore --configuration Release
-
-lint:
+.dotnet-sdk-base:
+  cache:
+    key: "$CI_JOB_STAGE-$CI_COMMIT_REF_SLUG"
```

Comment:
can you just make this a cache key based on the project assets?
```
cache:
  key:
    file: /packages/pangea-sdk/PangeaCyber.Net/obj/project.assets.json
```

you can do this if a change to that file implies a change to dependencies/things that are downloaded during the ci run